src/du_parameters/frequency.py

```diff
@@ -0,0 +1,569 @@
+        Args:
+            other (Any): The value to divide by.
+
+        Returns:
+            GSCN: A new GSCN instance with the updated channel.
+
+        Raises:
+            NotImplementedError: If the other value is not an GSCN, int or Decimal.
+        """
+        if isinstance(other, GSCN):
+            return GSCN(round(self._channel / other._channel))  # type: ignore[operator]
+        if isinstance(other, int | Decimal):
+            return GSCN(round(self._channel / other))  # type: ignore[operator]
+        raise NotImplementedError(f"Unsupported type for division: {type(other).__name__}")
+
+    @staticmethod
+    def to_frequency(gscn: "GSCN") -> Frequency:
+        """Calculate the frequency using input GSCN.
+
+        Args:
+           gscn (GSCN): The input GSCN.
+
+        Returns:
+            frequency (Frequency): The closest frequency.
+
+        Raises:
+            ValueError: If the GSCN is out of supported range or n is out of range.
+        """
+        config = get_range_from_gscn(gscn)
+        if config.name == "LowFrequency":
+            # Special calculation for low frequencies with scaling factor (m_scaling)
+            n = gscn / CONFIG_CONSTANT_THREE
+            if is_valid_n(n, config.min_n, config.max_n):  # type: ignore[operator]
+                result = (
+                    n._channel * config.multiplication_factor  # type: ignore[operator]
+                    + config.m_multiplication_factor * config.m_scaling
+                )
+                return Frequency(result)
+
+            raise ValueError(
+                f"Value of N: {n} is out of supported range ({config.min_n}-{config.max_n})."
+            )
+
+        elif config.name in {"MidFrequency", "HighFrequency"}:
+            # For high/medium range frequencies
+            n = gscn - config.base_gscn
+            if is_valid_n(n._channel, config.min_n, config.max_n):  # type: ignore
+                result = config.multiplication_factor * n._channel + config.base_freq  # type: ignore
+                return Frequency(result)
+
+            raise ValueError(
+                f"Value of N: {n} is out of supported range ({config.min_n}-{config.max_n})."
+            )
+
+        raise ValueError(f"Unsupported configuration name: {config.name}")
+
+    @staticmethod
+    def from_frequency(frequency: Frequency) -> "GSCN":
+        """Calculate the closest GSCN for a given frequency.
+
+        Args:
+            frequency (Frequency): The input frequency.
+
+        Returns:
+            GSCN (GSCN): The closest GSCN.
+
+        Raises:
+            ValueError: If the frequency is out of supported range or n is out of range.
+        """
+        config = get_range_from_frequency(frequency)
+        if config.name == "LowFrequency":
+            n = (
+                frequency - (config.m_scaling * config.m_multiplication_factor)
+            ) / config.multiplication_factor
+            if is_valid_n(n, config.min_n, config.max_n):
```

Comment:
In this case, `n` type would be `Frequency`. Should we cast it do Decimal for consistency?